pkg/volume/util/util.go

```diff
@@ -608,6 +609,27 @@ func GetPodVolumeNames(pod *v1.Pod) (mounts sets.String, devices sets.String) {
 	return
 }
 
+// FsUserFrom returns FsUser of pod, which is determined by the runAsUser
+// attributes.
+func FsUserFrom(pod *v1.Pod) *int64 {
+	var fsUser *int64
+	// Exclude ephemeral containers because SecurityContext is not allowed.
+	podutil.VisitContainers(&pod.Spec, podutil.InitContainers|podutil.Containers, func(container *v1.Container, containerType podutil.ContainerType) bool {
+		runAsUser, ok := securitycontext.DetermineEffectiveRunAsUser(pod, container)
+		// One container doesn't specify user or there are more than one
+		// non-root UIDs.
```

Comment:
I think there might be a couple problems with this logic, if I'm reading it correctly:

1. A container could set the user in the container image. To get the image user, you can use the [`getImageUser(...)`](https://github.com/kubernetes/kubernetes/blob/205d5c58296357365bfa834ecabb384c1c1042c3/pkg/kubelet/kuberuntime/helpers.go#L122) function, but you'd need to plumb through the runtime and export the method. Also note that the username can't be depended on.
2. A root user can only be ignored if it also has CAP_DAC_OVERRIDE (although it's included by default).

Even with these precautions, it's still possible that the container could drop permissions at runtime, but I think it should be acceptable to just cover that case by documentation (i.e. use FSGroup)

Reply:
Actually, looks like I was misreading the code... I think this says that if the user isn't explicitly specified in all the containers (and the explicit users don't match) then return `nil`. In that case, I think the code is correct but the comment is incorrect (since it doesn't care about non-root).

Reply:
using `getImageUser` would require a lot of changes in kubelet in terms of the image pulling and volume mounts.